services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceControllerService.java

```diff
@@ -106,7 +106,8 @@ public VeniceControllerService(
     KafkaPubSubMessageDeserializer pubSubMessageDeserializer = new KafkaPubSubMessageDeserializer(
         kafkaValueSerializer,
         new LandFillObjectPool<>(KafkaMessageEnvelope::new),
-        new LandFillObjectPool<>(KafkaMessageEnvelope::new));
+        new LandFillObjectPool<>(KafkaMessageEnvelope::new),
+        new PubSubTopicRepository()); // TODO: Reuse PubSubTopicRepository we pass into this class if needed.
```

Comment:
We do have a `pubSubTopicRepository` instantiated already just a few lines above this one, so if we must pass one here, then we should pass the same one... but as I said in another comment, it seems weird to require this class to get a `PSTR` passed during construction just for the sake of carrying it around. I would prefer if we removed it from here.